src/features/auto-resize-grid/js/auto-resize.js

```diff
@@ -19,25 +19,23 @@
       require: 'uiGrid',
       scope: false,
       link: function($scope, $elm, $attrs, uiGridCtrl) {
-
-        $scope.$watch(function() {
-          return $elm[0].clientWidth;
-        }, function(newVal, oldVal) {
-          if (newVal !== oldVal) {
-            uiGridCtrl.grid.gridWidth = newVal;
-            uiGridCtrl.grid.refresh();
+        $scope.$watchGroup([
+          function() {
+            return gridUtil.elementWidth($elm);
+          },
+          function() {
+            return gridUtil.elementHeight($elm);
+          }], function(newValues, oldValues, scope) {
+          if (newValues.toString() !== oldValues.toString()) {
+            uiGridCtrl.grid.gridWidth = newValues[0];
+            uiGridCtrl.grid.gridHeight = newValues[1];
+            setTimeout(function(){
```

Comment:
Whys is a timeout needed here? Specially if the time is set to 0.

Reply:
Here they talk about it more. https://stackoverflow.com/questions/22346990/why-is-using-ifscope-phase-scope-apply-an-anti-pattern

If I called scope apply directly, it offten happens that I saw error in javascript console because apply was called in middle of another digest cycle. Which lead to error in javascript. So when I was looking to other solutions. This looked like best one.